smetrics/src/test/scala/com/evolutiongaming/smetrics/MeasureDurationSpec.scala

```diff
@@ -24,6 +30,48 @@ class MeasureDurationSpec extends AnyFunSuite with Matchers {
     duration shouldEqual 2.nanos
     state shouldEqual State.Empty
   }
+
+  test("MeasureDurationOps.measured") {
+    val handler: FiniteDuration => StateT[Unit] =
+      duration => StateT { state =>
+        val timestamp = duration.toNanos
+        timestamp shouldEqual 5
+        (state, timestamp)
+      }.void
+    val (state, _) = Timer[StateT]
+      .sleep(5.nanos)
+      .measured(handler)
+      .run(State.Empty)
+    state shouldEqual State.Empty
+  }
+
+  test("MeasureDurationOps.measuredCase success") {
+    val successHandler = mock[FiniteDuration => IO[Unit]]
```

Comment:
Please remove mockito. I see you have `Timer[StateT]`, why did you not use it instead of `Timer[IO]`, which actually sleeps, as I understand?

Reply:
measuredCase required MonadError instance while cats.data.IndexedStateT provides method to generate Monad instance there are no similar one for MonadError and I can not found how to instantiate MonadError[StateT]

Reply:
Change the effect to `StateT[Either[Throwable, *], State, *]`? As I see there is only one test except yours, so not much to change

Reply:
Or, better yet, does not fix anything for the effect, and make `timerStateT` abstract on the inner F. So in the old test you'll change nothing, and in new ones will use `Timer[StateT[Either[Throwable, *]], FiniteDuration, *]`